smt/surrogate_models/krg_based.py

```diff
@@ -1062,6 +1062,10 @@ def grad_minus_reduced_likelihood_function(log10t):
             self.D = self._componentwise_distance(D, opt=ii)
 
             # Initialization
+            try:
+                del optimal_theta_res
```

Comment:
I think that deleting a variable is definitely a code smell. We certainly need to refactor the whole thing to avoid keeping an undesired state.

Reply:
Yes, I agree, the variables should not be keep in memory. It causes problem for KPLSK because we do two optimization cake some "trash" from the first interact with the second one.